### PR TITLE
Remove more projections code

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit e0d469b0e06b5592dc5fb80ca93f52afff579db6 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58628/e0d469b0e06b5592dc5fb80ca93f52afff579db6/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58628/e0d469b0e06b5592dc5fb80ca93f52afff579db6/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by Algunenano — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp [hunk 1]
```diff
@@ -135,9 +135,6 @@ QueryPlanPtr MergeTreeDataSelectExecutor::read(
     std::shared_ptr<PartitionIdToMaxBlock> max_block_numbers_to_read,
     bool enable_parallel_reading) const
 {
-    if (query_info.merge_tree_empty_result)
```
Comment (Algunenano) [inline src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp:138]: Nice. This is the kind of things static analysis should handle :sweat:

**Comment by CurtizJ** [comment]
Test `00992_system_parts_race_condition_zookeeper_long` has been failed in master too [link](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gSU5URVJWQUwgMjQgSE9VUgogICAgQU5EIHB1bGxfcmVxdWVzdF9udW1iZXIgPSAwCiAgICBBTkQgdGVzdF9zdGF0dXMgIT0gJ1NLSVBQRUQnCiAgICBBTkQgdGVzdF9zdGF0dXMgTElLRSAnRiUnCiAgICBBTkQgY2hlY2tfc3RhdHVzICE9ICdzdWNjZXNzJwogICAgQU5EIHBvc2l0aW9uKHRlc3RfbmFtZSwgJzAwOTkyX3N5c3RlbV9wYXJ0c19yYWNlX2NvbmRpdGlvbl96b29rZWVwZXJfbG9uZycpID4gMApPUkRFUiBCWSBjaGVja19zdGFydF90aW1l)

**Comment by CurtizJ** [comment]
Performance tests: `parallel_index`

**Pull request merged**